### ISSUE TITLE
update jsrrdgraph to new rrdtool

### ISSUE THREAD

**Issue by xfiles27** [manuelluis/jsrrdgraph#25]
An interesting idea, are there plans to develop the project further? Under the new version of the RRD, more complex CDEF functions, STACK support etc ?

**Comment by manuelluis** [comment]
The initial idea was do the render of the graph in the browser translating the C code to javascript. The use of javascriptRRD (to get the rrdfiles) was only for testing I was hopping that rrdtool would come with some rest interface to get the data but this has not happened.

I really like the look of the rrdtools graphs, but with the domination of grafana seems complicated to see more tools using graphs with the look of rrdtool.

I there is some interest, I have some ideas:
- use WASM to increase the speed
- make the svg graph more interactive: show data when select a line or area.
- define a yaml/json to represent the graph instead of the rrdtool syntax.
- diverge from rrdtool graph. For example, the areas: you cannot specify the color of the line and the color of the area at the same time, rrdtool is painting a line and a area and this complicates the interactivity of the graph.

Also there is some libraries in javascript that use d3.js to graphs with the look and feel of rrdgraph, like:
https://github.com/oetiker/RrdGraphJS

The add of more CDEF functions it's not complicated.
The STACK was not added because it was deprecated.

**Comment by gvanem** [comment]
I have an issue with the generated `test/data/*.rrd` files. Running e.g.
`rrdtool info test\data\cpu-idle.rrd` prints:
```
ERROR: 'test\data\cpu-idle.rrd' is too small (should be -858976496 bytes)
```

I see that RRD-version is correct `0003`, but not sure about the other header-values.
Is this a structure packing issue or an endian issue?

**Comment by manuelluis** [comment]
This tests files was created some time ago in a 32bit machine.
The javascript code handles architecure differences.

```
$ file ./usr/bin/rrdtool                
usr/bin/rrdtool: ELF 32-bit LSB pie executable, Intel i386, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux.so.2, BuildID[sha1]=49684eea716ee93de77825fb15410171540f2cd2, for GNU/Linux 3.2.0, stripped
$ ./usr/bin/rrdtool info cpu-idle.rrd 
filename = "cpu-idle.rrd"
rrd_version = "0003"
step = 10
```

**Comment by gvanem** [comment]
> The javascript code handles architecure differences.

But the C-code in **rrdtool** seemingly does not. 

So is a `.rrd` file created on a 32-bit Linux machine compatible with Windows/x64 or not? If not, yikes!

**Comment by manuelluis** [comment]
No, it's not compatible, you need to do a `rrdtool dump` and `rrdtool restore` to move files between diferent architecures.
Also the representation of float point can also change between architectures.
